### PR TITLE
Add a build-hook feature to gulp-tasks

### PR DESCRIPTION
**Pull request by Lichtjaeger**
+ some repairs for the gulp-tasks

### CODE REVIEW

**Comment by Lichtjaeger** [comment]
Hi,
bozon is evolving and I like it.

I have updated my commit, so it's now consistend with your optimizations.

There is now a `bozon.buildTask()` next to `bozon.task()`.

File: tasks/build.js [hunk 1]
```diff
@@ -27,18 +27,18 @@ bozon.task('scripts:renderer', function () {
...
 bozon.task('images', function () {
-  bozon.src('images/**/*').pipe(bozon.dest('images'));
+  return bozon.src('images/**/*').pipe(bozon.dest('images'));
 });
 
-bozon.task('compile', function () {
+bozon.task('compile', function (callback) {
   settings = new bozon.Settings();
-  runSequence('scripts:main', 'scripts:renderer', 'styles', 'html', 'images', 'config');
+  runSequence.apply(this, ['scripts:main', 'scripts:renderer', 'styles', 'html', 'images'].concat(bozon.hooks, 'config', callback));
```
Comment (alchapone) [inline tasks/build.js]: I really like the idea of adding hooks to bozon gulp tasks, but not sure about placing them into this specific place in a sequence. Wouldn't it be more flexible to allow users to define where their tasks should take place? Maybe something like: 
```javascript
task = bozon.task('whatever', function () { ... });
bozon.insertAfter('images', task);
```

or even

```javascript
bozon.runAfter('images', function () { ... });
bozon.runBefore('config', function () { ... });
```

Comment (Lichtjaeger) [inline tasks/build.js]: If you would like a taskplacement I can add one.

I would prefere your second example. Something like:

- `bozon.buildTask(...)` - add the task to the end of pipeline (after config)
- `bozon.buildTaskAfter('x', 'taskname', ...)` - add the task after task "x"
- `bozon.buildTaskBefore('x', 'taskname', ...)` - add the task before task "x"

If "x" is not defined then default to add the task to the end.

Is this ok for you?

File: lib/bozon.js [hunk 2]
```diff
@@ -11,6 +11,23 @@ var bozon = {
     return gulp.task.apply(gulp, arguments);
   },
 
+  buildTask: function () {
+    bozon.hooks.push(arguments[0]);
```
Comment (alchapone) [inline lib/bozon.js]: For consistency sake, could you use `this.hooks` instead of `bozon.hooks`?

Comment (Lichtjaeger) [inline lib/bozon.js]: done

**Comment by alchapone** [comment]
Ok, cool. Thanks @Lichtjaeger !

**Pull request merged**